src/librustc_data_structures/sync.rs

```diff
@@ -730,9 +730,9 @@ impl<T> OneThread<T> {
     }
 
     #[inline(always)]
-    pub fn into_inner(value: Self) -> T {
-        value.check();
-        value.inner
+    pub fn into_inner(self) -> T {
```

Comment:
That's not quite the same thing. Now you can call this via `value.into_inner()`. I'm presuming the original author wanted it to be called as `T::into_inner(value)`.

Reply:
I didn't get any errors when running stage 1 tests, so it might not even be used.

Reply:
`T::into_inner(value)` would still work after your change, but now `value.into_inner()` would also work, which is something we try to avoid in certain situations.

Yea, please revert it.